lib/embroider/update-manifest-from-html.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const fs = require('fs');
+const findScriptSrcs = require('find-scripts-srcs-in-document');
+
+function getFileName(filePath) {
+  const scriptFileNameRegEx = /([a-zA-Z0-9_\.\-\(\):])+(\.js)/ig;
+  return Array.isArray(filePath.match(scriptFileNameRegEx)) ? filePath.match(scriptFileNameRegEx)[0] : '';
+}
+
+module.exports = function updateManifestFromHtml(root) {
+  // package.json is not present in root location skip
+  // below flow as there is no package.json to update.
+  if (!fs.existsSync(root + '/package.json')) {
+    return;
+  }
+
+  const pkg = JSON.parse(fs.readFileSync(root + '/package.json', 'UTF8'));
+  const appFile = pkg.fastboot.manifest.appFiles[0];
+
+  // if app.js file exists skip reading from the index.html as
+  // it is classic build flow
+  if(appFile && fs.existsSync(`${root}${appFile}`)) {
+    return;
+  }
+
+  // otherwise, we must parse the index.html file to figure out what's next
+  const indexHtml = fs.readFileSync(root + '/index.html', 'UTF8')
+  // TODO: fix simple-html-tokenizer: https://github.com/tildeio/simple-html-tokenizer/pull/71
+    .replace('<!DOCTYPE html>','');
+
+  const assetsDir = `assets/`;
+  const vendorFiles = pkg.fastboot.manifest.vendorFiles;
+
+  // Find script src which are not data-fastboot-ignore and data-embroider-ignore
+  const indexScriptFilePaths = findScriptSrcs(indexHtml, scriptToken => {
+    const { attributes } = scriptToken;
+    return attributes.find(attr => attr[0] === 'data-embroider-ignore' || 'data-fastboot-ignore');
+  }).filter( src => {
+    // skipping files if they are part of vendorFiles list
+    // this would cover any vendor files (vendor or vendor-static)
+    const fileName = getFileName(src);
+    return !vendorFiles.find(filePath => fileName === getFileName(filePath));
+  });
+
+  // Building appFiles list from html.
+  const appFiles = [];
+  indexScriptFilePaths.forEach(src => {
+    // Extract the file name from the src file paths and
+    // checks the file exits in the assets location on the disk.
+    const fileName = getFileName(src);
+    const assetPath = `${assetsDir}${fileName}`;
+    const filePath = `${root}/${assetPath}`;
+    if(fileName && !appFiles.includes(assetPath) && fs.existsSync(filePath)) {
+      appFiles.push(assetPath);
+    }
+  });
+
+  const fastbootFile = pkg.fastboot.manifest.appFiles.find(x => /-fastboot.js$/.test(x))
+  if (!fastbootFile) {
+    throw new Error(`ember-cli-fastboot: missing '<app-name>-fastboot.js' file`)
```

Comment:
Are their common problems that can cause this? Is there something the user can do to address this issue?

Reply:
Actually, the only time this file will not be available if package.json generation had issues during the build. I kept this check when I pulled in your changes. 

I think we don't need that check, as the expectation after installing ember-cli-fastboot is that the proper package.json is generated during the build time. Let me remove that check.

Reply:
I removed this check, as `ember-cli-fastboot` will always generate the package.json with `<appname>-fastboot.js`.